### PR TITLE
Add baseline profile generation

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Add module for generating a [baseline profile](https://developer.android.com/topic/performance/baselineprofiles/overview) for the app.

### CODE REVIEW

**Comment by IndusAryan** [comment]
wow, pr so advanced that i couldn't understand what the code is doing (ik making app fast somehow) and the pixel props included, maybe @RafaelsRamos can drop his 2 cents here with good explanation.

**Comment by Isira-Seneviratne** [comment]
> wow, pr so advanced that i couldn't understand what the code is doing (ik making app fast somehow) and the pixel props included (for what?), maybe @RafaelsRamos can drop his 2 cents here with good explanation.

It allows a profile of methods called during startup to be generated, so that they can be compiled ahead of time to improve startup performance on Android 7.0 and later.

File: baselineProfile/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import com.android.build.api.dsl.ManagedVirtualDevice
+
+plugins {
+    alias(libs.plugins.androidTest)
+    alias(libs.plugins.jetbrainsKotlinAndroid)
+    alias(libs.plugins.baselineprofile)
+}
+
+android {
+    namespace = "com.example.baselineprofile"
```
Comment (RafaelsRamos) [inline baselineProfile/build.gradle.kts:10]: Should we update the module namespace?

```suggestion
    namespace = "com.github.libretube.baselineprofile"
```

**Comment by IndusAryan** [comment]
i understood now, but why pixel virtual device prop is needed for tests and why source compat of baseline is not jdk 17 (LTS) but 1_8 ,different from Libretube

**Comment by RafaelsRamos** [comment]
The source compact can be `17`, it defaulted to `1_8` because Android Studio generated it this way (it is probably a good idea to change it to `17` for consistency).

Regarding the virtual device, the way the baseline profile generates the profiles, is by pre-compiling classes and methods from flows we want him to optimize (startup on this PR). This must be done on rooted devices (rooted emulators or physical devices).

The reason why we want it on an emulator is because it should be ran before a new release, so adding it to the release pipeline would be the ultimate goal.

**Review by Bnyro — APPROVED** [review]

**Pull request merged**